### PR TITLE
feat: add process list and list item

### PR DESCRIPTION
**Pull request by storyworks**
Features:

**Process lists**
- All Contacts for a user are displayed and also grouped by process type
- If only 1 process type exists, only show All tab
- No tab sorting order (Tatai said it's not MVP)
- Alphabetical contact sorting order within tabs (Ed originally wanted "recently added" aka when the process was assigned to them. But this is not possible yet so going with alphabetical)
- Search by contact name filters process cards on each tab (Additional requirement from Tatai)

**ProcessListItem**
- Displaying name, process types and number of incomplete/pending tasks. No other info is needed.
- Clicking on the card redirects user to PersonDetailPage

https://user-images.githubusercontent.com/10802634/180121023-f06f8bf1-d3ff-421c-b2b1-39e25fde941e.mov

### CODE REVIEW

File: src/lib/queries/getContact/getContact.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export interface GetContact {
   lastName: string
   phoneNumbers: string[]
   emails: string[]
-  process: {
+  processes: {
```
Comment (storyworks) [inline src/lib/queries/getContact/getContact.ts:9]: Doesn't need to be tied to fluro's naming conventions so renamed for clarity

File: src/@types/fluro.d.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ declare module 'fluro' {
     verified: boolean
     account: Account
     permissionSets: { [key: string]: PermissionSet }
+    contacts: string[]
```
Comment (storyworks) [inline src/@types/fluro.d.ts:27]: Added field here for now, after discussion with Tatai - User will be moved to the useAuth file in a later refactor.

**Comment by storyworks** [comment]
TODO: 

Display in each card - Name, process type, X tasks (across each process type from 'incomplete and pending')

File: src/components/PersonListItem/PersonListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  const processLabel =
+    processes.length === 0
+      ? ''
+      : processes.length === 1
+      ? processes[0]
+      : processes.length === 2
+      ? `${processes.slice(0, -1).join(', ')} & ${processes.slice(-1)}`
+      : `${processes.slice(0, 2).join(', ')} & ${
+          processes.slice(2).length
+        } more`
```
Comment (tataihono) [inline src/components/PersonListItem/PersonListItem.tsx:29]: This was pretty hard to understand what was going on. Breaking into a switch statement would make it clearer. Also, there was no test for the case where processes are zero.

```suggestion
  let processLabel: string
  switch (processes.length) {
    case 0:
      processLabel = ''
      break
    case 1:
      processLabel = processes[0]
      break
    case 2:
      processLabel = `${processes.slice(0, -1).join(', ')} & ${processes.slice(
        -1
      )}`
      break
    default:
      processLabel = `${processes.slice(0, 2).join(', ')} & ${
        processes.slice(2).length
      } more`
  }
```

File: src/components/PersonListItem/PersonListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    <Card sx={{ width: '100%', maxWidth: 345, mb: 2 }}>
+      <CardActionArea href={`/tabs/people/${contact._id}`}>
+        <CardContent>
+          <Typography variant="caption" color="text.secondary">
+            {processLabel}
+          </Typography>
+          <Typography
+            variant="h6"
+            gutterBottom
+          >{`${contact.firstName} ${contact.lastName}`}</Typography>
```
Comment (tataihono) [inline src/components/PersonListItem/PersonListItem.tsx:49]: Don't think you need to have a special variable js string here.
```suggestion
          <Typography variant="h6" gutterBottom>
            {contact.firstName} {contact.lastName}
          </Typography>
```

File: src/lib/queries/getContacts/getContacts.handlers.ts [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+export function getProcessesHandler(): RestHandler {
+  return rest.post('https://api.fluro.io/content/_query', (req, res, ctx) => {
+    req.body = {
+      types: 'process',
+      assignedTo: { $in: ['userContactId'] }
+    }
+    return res(
+      ctx.status(200),
+      ctx.json([
```
Comment (tataihono) [inline src/lib/queries/getContacts/getContacts.handlers.ts:11]: type this response
```suggestion
      ctx.json<GetProcesses>([
```

File: src/lib/queries/getContacts/getContacts.handlers.ts [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+export function getProcessesHandlerSimple(): RestHandler {
+  return rest.post('https://api.fluro.io/content/_query', (req, res, ctx) => {
+    req.body = {
+      types: 'process',
+      assignedTo: { $in: ['userContactId'] }
+    }
+    return res(
+      ctx.status(200),
+      ctx.json([
```
Comment (tataihono) [inline src/lib/queries/getContacts/getContacts.handlers.ts:81]:
```suggestion
      ctx.json<GetProcesses>([
```

File: src/lib/queries/getContacts/getContacts.handlers.ts [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+  return rest.post(
+    'https://api.fluro.io/content/contact/multiple',
+    (req, res, ctx) => {
+      req.body = {
+        ids: ['aa', 'bb', 'cc', 'ck', 'dd'],
+        select: ['firstName', 'lastName']
+      }
+      return res(
+        ctx.status(200),
+        ctx.json([
```
Comment (tataihono) [inline src/lib/queries/getContacts/getContacts.handlers.ts:132]:
```suggestion
        ctx.json<GetContacts>([
```

File: src/pages/PeoplePage/PersonListPage/PersonListPage.tsx [hunk 8]
```diff
@@ -1,27 +1,21 @@
...
     <IonPage>
       <IonHeader>
-        <PeopleMenuBar onChange={(value): void => setSearch(value)} />
+        <PeopleMenuBar onChange={setSearch} />
       </IonHeader>
       <IonContent fullscreen>
-        {search}
-        {processDefinition != null &&
-          Object.keys(processDefinition).map((key) => <div>{key}</div>)}
+        {user != null && <PersonList search={search} user={user} />}
```
Comment (tataihono) [inline src/pages/PeoplePage/PersonListPage/PersonListPage.tsx:18]: this page is never rendered where user is null.

File: src/pages/PeoplePage/PersonListPage/PersonListPage.tsx [hunk 9]
```diff
@@ -1,27 +1,21 @@
...
 import { ReactElement, useState } from 'react'
-import { useQuery } from 'react-query'
 
 import { PeopleMenuBar } from '../../../components/PeopleMenuBar'
-import { getProcessDefinitions } from '../../../lib/queries/getProcessDefinitions'
+import { PersonList } from '../../../components/PersonList'
+import { useAuth } from '../../../lib/useAuth'
 
 export function PersonListPage(): ReactElement {
+  const { user } = useAuth()
```
Comment (tataihono) [inline src/pages/PeoplePage/PersonListPage/PersonListPage.tsx:9]: this should be handled in the component?

**Review by tataihono — CHANGES_REQUESTED** [review]

File: src/components/PersonList/PersonList.tsx [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+        allProcesses.all.push(process.item)
+        allProcesses[process.definition].push(process.item)
+      })
+    })
+  }
+
+  return allProcesses
+}
+
+export function PersonList({ search, user }: PersonListProps): ReactElement {
```
[the remark this answers is not in the document]
Comment (storyworks) [reply src/components/PersonList/PersonList.tsx:49]: Mainly did this so I wouldn't need to mock Auth so much will change...

File: src/components/PersonList/PersonList.tsx [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+  const processTypes = Object.keys(allProcesses)
+  const oneProcessType = processTypes.length < 3
+
+  const handleChange = (e: React.SyntheticEvent, newValue: string): void => {
+    setProcessType(newValue)
+  }
+
+  return (
+    <TabContext value={processType}>
+      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
```
[the remark this answers is not in the document]
Comment (storyworks) [reply src/components/PersonList/PersonList.tsx:75]: A divider immediately after? I was following material documentation 
image
but can update

File: src/pages/PeoplePage/PersonListPage/PersonListPage.tsx (hunk 8 above)
Comment (storyworks) [reply src/pages/PeoplePage/PersonListPage/PersonListPage.tsx:18]: The page will be there but the list just won't show if there's no user. What should it display if there are no processes assigned to a user?

File: src/pages/PeoplePage/PersonListPage/PersonListPage.tsx (hunk 9 above)
Comment (storyworks) [reply src/pages/PeoplePage/PersonListPage/PersonListPage.tsx:9]: I was trying to extract it out to this page so we wouldn't need to mock Auth in so many places..

File: src/components/PersonList/PersonList.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (storyworks) [reply src/components/PersonList/PersonList.tsx:75]: image
Ah, the reason they use box is because the Divider leaves a small gap - I'm not keen to use negative margins to fix it, so going to leave it like the example

**Review by tataihono — APPROVED** [review]
